### ISSUE TITLE
Make last four collision_list functions arguments optional.

### ISSUE THREAD

**Issue by Dupletor** [YoYoGames/GameMaker-Bugs#4435]
### Is your feature request related to a problem?

Not directly to a problem, but it forces the user to create a ds_list even if it is never used and has to be constantly cleaned.

### Describe the solution you'd like

Collision_list functions are like:
collision_circle_list(x1, y1, rad, obj, prec, notme, list, ordered);

They should be like:
collision_circle_list(x1, y1, rad, obj, [prec]=false, [notme]=true, [list]=undefined, [ordered]=false);

And just not require a list that is not necessarily used afterwards, now you have a collision counter that gives you how many instances/tiles of the given type exist in the defined area.

Also, [prec] argument could be defined by the sprite_index attributed to the assets given to the function, I personally feel like these functions shouldn't need it, like instance_place doesn't need it.

### Additional context

These functions are very good, but they attribute to the user some actions that similar functions don't. They feel improvable and quite out of place.

**Comment by tabularelf** [comment]
Question is, why do you want an instance collision count?

**Comment by nullbunny** [comment]
I also would like this feature, I've made throwaway lists unused before a couple of times in my projects. Sometimes, all you want to do is count the number of instances in an area. This is such a basic data query I think it should be added.

**Comment by Dupletor** [comment]
> Question is, why do you want an instance collision count?

You want an instance collision count every time you use the return of the function, since that is what the collision_list functions return.

For example:
You have a platform and weights above it. You want it to go down if there are 4 or more weights above it, stay still with 3, and go up with 1 or less.

```
var weights_count = collision_line_list(x - sprite_xoffset, y - sprite_yoffset - 5, 
				  x - sprite_xoffset + sprite_width, y - sprite_yoffset - 5, 
				obj_Weight, false, true, weights_list, false);
```
There is a situation in which you need to know what instances the weights are, because the weights might be differently weighted, but in the case where all weights are the same, the four last arguments are meaningless.

**Comment by tabularelf** [comment]
I feel that having a separate function would be better than changing how the arguments for `collision_line_list` works.
Like `instance_count_rectangle()`/`instance_count_line()`/`instance_count_circle()` might make more sense here.

**Comment by Dupletor** [comment]
Absolutely no need for that, since the collision_line_list literally returns the instance count.

**Comment by tabularelf** [comment]
I'm aware, but I also don't like the idea of making the list "optional", when the function itself is literally called `collision_line_list`. lol

**Comment by attic-stuff** [comment]
i agree with the elf on this one: a function that works like place_meeting but returns the amount of collisions would be radical but changing the list function to no longer use a list would, imo, really muddy the waters on the function. you can also use a counter that increments using a collision event, since the collision events fire off for each collision.

**Comment by backYard321** [comment]
Personally I think the `collision_list` functions in general are kind of a mess! I'm not necessarily opposed to the changes you're recommending (frankly I think they're fairly inoffensive compared to having dedicated `collision_count` variants), but long-term I'm generally in favour of deprecating all of the list collision functions in favour of something that returns an array.

So for this case:
```gml
var weights_count = collision_line_list(x - sprite_xoffset, y - sprite_yoffset - 5, 
				  x - sprite_xoffset + sprite_width, y - sprite_yoffset - 5, 
				obj_Weight, false, true, weights_list, false);
```
you could instead do this:
```gml
var _collisions= collision_line_array(x - sprite_xoffset, y - sprite_yoffset - 5, 
				  x - sprite_xoffset + sprite_width, y - sprite_yoffset - 5, 
				obj_Weight, false, true, weights_list, false);
var weights_count  = array_length(_collisions)
```

**Comment by mistletoe** [comment]
> Personally I think the `collision_list` functions in general are kind of a mess! I'm not necessarily opposed to the changes you're recommending (frankly I think they're fairly inoffensive compared to having dedicated `collision_count` variants), but long-term I'm generally in favour of deprecating all of the list collision functions in favour of something that returns an array.
> 
> So for this case:
> 
> ```c++
> var weights_count = collision_line_list(x - sprite_xoffset, y - sprite_yoffset - 5, 
> 				  x - sprite_xoffset + sprite_width, y - sprite_yoffset - 5, 
> 				obj_Weight);
> ```
> 
> you could instead have this:
> 
> ```c++
> var _collisions= collision_line_array(x - sprite_xoffset, y - sprite_yoffset - 5, 
> 				  x - sprite_xoffset + sprite_width, y - sprite_yoffset - 5, 
> 				obj_Weight);
> var weights_count  = array_length(_collisions)
> ```

+1; having arrays that get automatically GC'd when no longer needed because they're just a temp-var would be nice. 

In general, I wish _anything_ in a temp-var that wasn't then put into a Struct, Instance Variable or Global Variable or passed forwards as a parameter for another function was always GC'd at the end of functions, getting rid of 90% of the drudgery involved w/ "manually-allocated resources" that we know we're just deleting at the end of function calls.
